### PR TITLE
Fix out of bounds error in pid_tuning

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes two index out of bounds errors in PID Tuning tab.

### CODE REVIEW

**Comment by haslinghuis** [comment]
Sonar is totally wrong here. The following code would result in out of bound error:
```
for (balloon of balloons) {
    balloon();
}
```

**Review by mikeller — DISMISSED** [review]

File: src/js/tabs/pid_tuning.js [hunk 1]
```diff
@@ -2412,7 +2412,7 @@ TABS.pid_tuning.updateRatesLabels = function() {
                 );
             }
             // then display them on the chart
-            for (let i = 0; i <= balloons.length; i++) {
+            for (let i = 0; i < balloons.length; i++) {
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js:2415]: These _could_ easily be done as `for ( of )`:

```
for (balloon of balloons) {
    balloon.balloon();
}
```

same for the above.

Comment (haslinghuis) [reply src/js/tabs/pid_tuning.js:2415]: 😜 I've tried but then the last element is still undefined.

image

Comment (haslinghuis) [reply src/js/tabs/pid_tuning.js:2415]: Let me change it - we both forgot to initialize the balloon 🤣

Comment (haslinghuis) [reply src/js/tabs/pid_tuning.js]: Tested and fixed

**Review by McGiverGim — APPROVED** [review]

**Review by mikeller — APPROVED** [review]

**Pull request merged**